### PR TITLE
Setup SG3 CI system

### PR DESCRIPTION
**Pull request by leifmadsen**
Setup the SG3 CI system so that it passes the new CI interfaces.

### CODE REVIEW

**Comment by leifmadsen** [comment]
[Add sg-core and sg-build + validate_deployment.sh updates + smoketest adjustments to latest CI scripts](https://trello.com/c/Wk9OR2wR/1062-add-sg-core-and-sg-build-validatedeploymentsh-updates-smoketest-adjustments-to-latest-ci-scripts)

**Comment by leifmadsen** [comment]
Results in the following Smart Gateway `Deployment` manifests:

```
for i in $(oc get deployments.apps | grep smartgateway | cut -f1 -d' '); do oc get deployment $i -oyaml; echo "----------------------------------" ; echo; echo; done
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2020-07-13T16:39:25Z"
  generation: 1
  name: stf-default-ceilometer-notification-smartgateway
  namespace: service-telemetry
  ownerReferences:
  - apiVersion: smartgateway.infra.watch/v2alpha1
    kind: SmartGateway
    name: stf-default-ceilometer-notification
    uid: 5b475c62-2e52-4c98-8c68-da79df9b1438
  resourceVersion: "5151258"
  selfLink: /apis/apps/v1/namespaces/service-telemetry/deployments/stf-default-ceilometer-notification-smartgateway
  uid: ec5c04ba-867c-4f26-be9f-14e936a4dfd6
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: smart-gateway
      smart-gateway: stf-default-ceilometer-notification
  strategy:
    rollingUpdate:
      maxSurge: 50%
      maxUnavailable: 50%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: smart-gateway
        smart-gateway: stf-default-ceilometer-notification
    spec:
      containers:
      - args:
        - -config=/config/smartgateway_config.json
        - -uname=$(MY_POD_NAME)
        - -servicetype=events
        env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        image: image-registry.openshift-image-registry.svc:5000/service-telemetry/smart-gateway:latest
        imagePullPolicy: Always
        name: smart-gateway
        ports:
        - containerPort: 8081
          name: events
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /config
          name: sg-config
        - mountPath: /config/certs
          name: elastic-certs
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: stf-default-ceilometer-notification-smartgateway
        name: sg-config
      - name: elastic-certs
        secret:
          defaultMode: 420
          secretName: elasticsearch-es-cert
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2020-07-13T16:39:28Z"
    lastUpdateTime: "2020-07-13T16:39:28Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2020-07-13T16:39:25Z"
    lastUpdateTime: "2020-07-13T16:39:28Z"
    message: ReplicaSet "stf-default-ceilometer-notification-smartgateway-85468575d"
      has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
----------------------------------


apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2020-07-13T16:38:34Z"
  generation: 1
  name: stf-default-ceilometer-telemetry-smartgateway
  namespace: service-telemetry
  ownerReferences:
  - apiVersion: smartgateway.infra.watch/v2alpha1
    kind: SmartGateway
    name: stf-default-ceilometer-telemetry
    uid: d0bb2098-3068-40e4-8b41-b980b4439cdf
  resourceVersion: "5150944"
  selfLink: /apis/apps/v1/namespaces/service-telemetry/deployments/stf-default-ceilometer-telemetry-smartgateway
  uid: 9054af42-aa2e-45de-985c-8507e3e15906
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: smart-gateway
      smart-gateway: stf-default-ceilometer-telemetry
  strategy:
    rollingUpdate:
      maxSurge: 50%
      maxUnavailable: 50%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: smart-gateway
        smart-gateway: stf-default-ceilometer-telemetry
    spec:
      containers:
      - args:
        - -config=/config/smartgateway_config.json
        - -uname=$(MY_POD_NAME)
        - -servicetype=metrics
        env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        image: image-registry.openshift-image-registry.svc:5000/service-telemetry/smart-gateway:latest
        imagePullPolicy: Always
        name: smart-gateway
        ports:
        - containerPort: 8081
          name: metrics
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /config
          name: sg-config
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: stf-default-ceilometer-telemetry-smartgateway
        name: sg-config
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2020-07-13T16:38:44Z"
    lastUpdateTime: "2020-07-13T16:38:44Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2020-07-13T16:38:34Z"
    lastUpdateTime: "2020-07-13T16:38:44Z"
    message: ReplicaSet "stf-default-ceilometer-telemetry-smartgateway-5c9dbb67fb"
      has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
----------------------------------


apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2020-07-13T16:39:05Z"
  generation: 1
  name: stf-default-collectd-notification-smartgateway
  namespace: service-telemetry
  ownerReferences:
  - apiVersion: smartgateway.infra.watch/v2alpha1
    kind: SmartGateway
    name: stf-default-collectd-notification
    uid: c09e4b6c-b5d2-434d-9485-80022b070a6f
  resourceVersion: "5151111"
  selfLink: /apis/apps/v1/namespaces/service-telemetry/deployments/stf-default-collectd-notification-smartgateway
  uid: e1caa28a-65f6-484a-81df-09da5f8be97c
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: smart-gateway
      smart-gateway: stf-default-collectd-notification
  strategy:
    rollingUpdate:
      maxSurge: 50%
      maxUnavailable: 50%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: smart-gateway
        smart-gateway: stf-default-collectd-notification
    spec:
      containers:
      - args:
        - -config=/config/smartgateway_config.json
        - -uname=$(MY_POD_NAME)
        - -servicetype=events
        env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        image: image-registry.openshift-image-registry.svc:5000/service-telemetry/smart-gateway:latest
        imagePullPolicy: Always
        name: smart-gateway
        ports:
        - containerPort: 8081
          name: events
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /config
          name: sg-config
        - mountPath: /config/certs
          name: elastic-certs
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: stf-default-collectd-notification-smartgateway
        name: sg-config
      - name: elastic-certs
        secret:
          defaultMode: 420
          secretName: elasticsearch-es-cert
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2020-07-13T16:39:08Z"
    lastUpdateTime: "2020-07-13T16:39:08Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2020-07-13T16:39:05Z"
    lastUpdateTime: "2020-07-13T16:39:08Z"
    message: ReplicaSet "stf-default-collectd-notification-smartgateway-5f67d94f54"
      has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
----------------------------------


apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2020-07-13T16:38:19Z"
  generation: 1
  name: stf-default-collectd-telemetry-smartgateway
  namespace: service-telemetry
  ownerReferences:
  - apiVersion: smartgateway.infra.watch/v2alpha1
    kind: SmartGateway
    name: stf-default-collectd-telemetry
    uid: bc644f7f-fa03-4612-bb27-bc259f2ea90b
  resourceVersion: "5150847"
  selfLink: /apis/apps/v1/namespaces/service-telemetry/deployments/stf-default-collectd-telemetry-smartgateway
  uid: 82b4934c-7007-4703-a5e5-f31ce7362ad4
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: smart-gateway
      smart-gateway: stf-default-collectd-telemetry
  strategy:
    rollingUpdate:
      maxSurge: 50%
      maxUnavailable: 50%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: smart-gateway
        smart-gateway: stf-default-collectd-telemetry
    spec:
      containers:
      - args:
        - --amqp_url
        - amqp://stf-default-interconnect.service-telemetry.svc.cluster.local:5673/collectd/telemetry
        - --verbose
        - --block
        - --stat_period
        - "60"
        - --rbc
        - "15000"
        image: image-registry.openshift-image-registry.svc:5000/service-telemetry/sg-bridge:latest
        imagePullPolicy: Always
        name: bridge
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /tmp
          name: socket-dir
      - args:
        - -promhost
        - 0.0.0.0
        - unix
        image: image-registry.openshift-image-registry.svc:5000/service-telemetry/sg-core:latest
        imagePullPolicy: Always
        name: smart-gateway
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /tmp
          name: socket-dir
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - emptyDir: {}
        name: socket-dir
      - configMap:
          defaultMode: 420
          name: stf-default-collectd-telemetry-smartgateway
        name: sg-config
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2020-07-13T16:38:31Z"
    lastUpdateTime: "2020-07-13T16:38:31Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2020-07-13T16:38:19Z"
    lastUpdateTime: "2020-07-13T16:38:31Z"
    message: ReplicaSet "stf-default-collectd-telemetry-smartgateway-b8f75c7df" has
      successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
----------------------------------
```

**Comment by leifmadsen** [comment]
No extra ConfigMap for the collectd telemetry SG:

```
oc get configmaps | grep smartgateway
stf-default-ceilometer-notification-smartgateway   1      43m
stf-default-ceilometer-telemetry-smartgateway      1      43m
stf-default-collectd-notification-smartgateway     1      43m
```

**Comment by leifmadsen** [comment]
Note that the CI actually passed without issue. The "failure" was in the after script section. Almost looks like the system disconnected just as the clean up was finishing. I'd not consider this :x: on the CI to be a blocker for merging.

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ language: python
 git:
   depth: 1
 install:
-- pip install operator-courier
+- pip install operator-courier==2.1.7
```
Comment (csibbitt) [inline .travis.yml:10]: infrawatch/service-telemetry-operator#108 :+1:

File: README.md [hunk 2]
```diff
@@ -46,8 +46,6 @@ The quickest way to start up Service Telemetry Framework for development is to
 run the `quickstart.sh` script located in the `deploy/` directory after starting
 up a [CodeReady Containers](https://github.com/code-ready/crc) environment.
 
-To deploy a local build of the Service Telemetry Operator itself, start by
```
Comment (csibbitt) [inline README.md:49]: :-1: Did something go wrong here? This doesn't look right.

File: build/stf-run-ci/tasks/main.yml [hunk 3]
```diff
@@ -5,29 +5,68 @@
...
-
-- name: Validate system is operational
-  shell: ./validate_deployment.sh
-  args:
-    executable: /bin/bash
-  register: validate_deployment
-
-- debug:
-    var: validate_deployment.stdout_lines
+- name: Set default image paths when skipping local builds
```
Comment (csibbitt) [inline build/stf-run-ci/tasks/main.yml:8]: AFAICT these are only used in the local builds, not at all when skipping them. Should they be moved into the block below?

```suggestion
- name: Set default image paths for local builds
```

File: build/stf-run-ci/tasks/preflight_checks.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Wait for Service Telemetry Operator to be Succeeded
+  shell: |
+    while ! oc get csv | grep service-telemetry-operator | grep Succeeded; do echo "waiting for Service Telemetry Operator..."; sleep 3; done
```
Comment (csibbitt) [inline build/stf-run-ci/tasks/preflight_checks.yml:4]: Is this subject to any built-in timeout when running locally, or just the global one from the CI system?

File: deploy/quickstart.sh [hunk 5]
```diff
@@ -1,106 +1,8 @@
...
-  installPlanApproval: Automatic
-  name: servicetelemetry-operator
-  source: infrawatch-operators
-  sourceNamespace: openshift-marketplace
-EOF
-while ! oc get csv | grep service-telemetry-operator | grep Succeeded; do echo "waiting for Service Telemetry Operator..."; sleep 3; done
-if [ ! -z "${KIND_SERVICETELEMETRY}" ]; then
-  oc create -f - <<< "${KIND_SERVICETELEMETRY}"
-fi
+ansible-playbook \
```
Comment (csibbitt) [inline deploy/quickstart.sh:5]: Around and around we go :smile:

File: tests/infrared/13/infrared-openstack.sh [hunk 6]
```diff
@@ -5,77 +5,120 @@ set -e
 #  VIRTHOST=my.big.hypervisor.net
```
Comment (csibbitt) [inline tests/infrared/13/infrared-openstack.sh:5]: Skipping a few files here which I can see are part of the ceilometer metrics smoketesting and not really part of this PR

File: tests/infrared/13/stf-connectors.yaml.template [hunk 7]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (csibbitt) [inline tests/infrared/13/stf-connectors.yaml.template:1]: Skipping as not part of the PR

File: tests/smoketest/ceilometer_publish.py [hunk 8]
```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env python
```
Comment (csibbitt) [inline tests/smoketest/ceilometer_publish.py:1]: Skipping as not part of the PR

File: tests/smoketest/smoketest_ceilometer_entrypoint.sh [hunk 9]
```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
```
Comment (csibbitt) [inline tests/smoketest/smoketest_ceilometer_entrypoint.sh:1]: Skipping as not part of the PR

File: README.md (hunk 2 above)
Comment (leifmadsen) [reply README.md:49]: Oh, I think I must have started to update this section to account for the new CI. Let me review and adjust.

File: build/stf-run-ci/tasks/main.yml (hunk 3 above)
Comment (leifmadsen) [reply build/stf-run-ci/tasks/main.yml:8]: These are used when skipping local builds, but when the builds have previously been done. The wording I agree is a bit awkward here.

The point of this is that you've done a local build, you've tried a deployment but the Ansible has failed for some reason, but the builds themselves worked fine. This allows you to skip waiting a few minutes for the builds to re-run when the code never changed for the builds.

Primarily this functionality is for the CI developer who wants to make changes in the workflow, test everything, but doesn't want to wait several minutes for local build images to re-build.

File: deploy/quickstart.sh (hunk 5 above)
Comment (leifmadsen) [reply deploy/quickstart.sh:5]: This reminds me I need to delete the `configs/` directory too :)

File: build/stf-run-ci/tasks/preflight_checks.yml (hunk 4 above)
Comment (leifmadsen) [reply build/stf-run-ci/tasks/preflight_checks.yml:4]: Just the global one from the CI system.

File: deploy/quickstart.sh (hunk 5 above)
Comment (leifmadsen) [reply deploy/quickstart.sh:5]: Oh maybe not, looks like it was already removed as part of the merge from master.

**Comment by leifmadsen** [comment]
@csibbitt ok round 2! I think I addressed the couple of minor things.

File: build/stf-run-ci/tasks/preflight_checks.yml (hunk 4 above)
Comment (csibbitt) [reply build/stf-run-ci/tasks/preflight_checks.yml:4]: Okay but the first time it annoys me I'm adding an extra timeout

**Comment by csibbitt** [comment]
:heavy_check_mark:
